apps/website/src/components/ComponentIcon.astro

```diff
@@ -0,0 +1,102 @@
+---
+import { renderJSX } from 'astro/dist/runtime/server';
+import fs from 'node:fs';
+
+export type Props = {
+  /** name of the file to read code example from, should be located in ~/examples folder */
+  icon: string;
+};
+
+const { icon } = Astro.props;
+
+// baseUrl refers to project root
+const baseUrl = import.meta.env.PROD
+  ? new URL('..', import.meta.url).href
+  : new URL('../../', import.meta.url);
+
+let code = '';
+
+if (icon) {
+  const codePath = new URL(`./src/icons/component-icons/${icon}.astro`, baseUrl);
+  code = await fs.promises.readFile(codePath, 'utf8');
+}
+
+// const componentName = icon.split('.')[0];
+// const capitalize = (s: string) => s.charAt(0).toUpperCase() + s.slice(1);
+---
+
+<div class='thumbnail' set:html={code} />
```

Comment:
I think you might not need to do all this extra work because [`Astro.glob`](https://docs.astro.build/en/guides/imports/#astroglob) can do it for you. Only catch is that it does not support dynamic strings so you need glob everything and then filter inside it.

```suggestion
const allThumbnails = await Astro.glob('../icons/component-icons/*.astro');
const thumbnail = allThumbnails.find(({ file }) => file.endsWith(`${icon}.astro`));
---

<thumbnail.default class='thumbnail' />
```